### DISCUSSION TITLE
Setting the loss % variables accurately

### DISCUSSION THREAD

**Discussion by pacemaker82** [springfall2008/batpred#779] [General]
I need help setting up my battery charge/discharge loss %, as well as inverter loss %. These are the current settings, as installed.

I have an GivEnergy AC3 with 9.5kWh battery

`input_number.predbat_battery_loss` set to 3%

`input_number.predbat_battery_loss_discharge` set to 3%

`input_number.predbat_inverter_loss` set to 4%

I dont think these are right for my system, but need help in setting the right % in the right places. Looking at my discharge data for a full discharge during the night - my battery pushes out 3300w, 2600w going to grid, 200w going to house as base load. Thats a 500w loss ~16% from 3100w discharge to 2600w to grid.

If I am interpreting the settings above correctly, the discharge loss + inverter loss variables should equal ~16% ?(vs the 7% it currently has set)?

If so, what balance is discharge loss vs inverter loss? 

FWIW 16% seems really high, but maybe I have incorrect expectations or approaching the calculations wrong

**Comment by springfall2008** [comment]
I don't think the calculation is quite that simple, firstly make sure you have hybrid turned off. Then using the calibration chart look at the real charging rate vs the predicted one (e.g h8 vs actual) if the slope on the curves align then you are about right. If the predicted charges faster than reality then you should add more losses. 

I would guess inverter loss of 7% maybe closer to reality for AC coupled but I'd be interested to see your findings.

![image](https://github.com/springfall2008/batpred/assets/48591903/51f5aaf4-eaf4-4095-a28e-8c657683ecdb)

**Comment by pacemaker82** [comment]
Here's my chart. Sometimes it looks like it follows exactly, sometimes wildly off... but this is first time I am reading it, maybe you can see it better :) 

<img width="914" alt="Screenshot 2024-02-19 at 19 12 34" src="https://github.com/springfall2008/batpred/assets/157808562/13ee1480-e9e2-40e2-8220-4e5c02ecf10e">

**Reply by SwiftRR** [reply]
I have a Givenergy AC3 connected to two GE 8.2 kWh batteries.
This is also set as 3%, 3%, 4% losses
This is my calibration curve. From previous comments, this looks pretty good to me.
<img width="362" alt="image" src="https://github.com/springfall2008/batpred/assets/68816364/6ca3156c-5443-4ee3-bf71-0f41b10b252f">

Rob
